docs/features/logging.rst

```diff
@@ -0,0 +1,189 @@
+
+========
+Handlers
+========
+
+The handler is the engine that determines what happens to each message in a logger. It describes a particular logging behavior, such as writing a message to the screen, to a file, or to a network socket.
+
+Like loggers, handlers also have a log level. If the log level of a log record doesn’t meet or exceed the level of the handler, the handler will ignore the message.
+
+=======
+Filters
+=======
+
+A filter is used to provide additional control over which log records are passed from logger to handler.
+
+==========
+Formatters
+==========
+
+Ultimately, a log record needs to be rendered as text. Formatters describe the exact format of that text. A formatter usually consists of a Python formatting string containing LogRecord attributes; however, you can also write custom formatters to implement specific formatting behavior.
+
+
+Here's an example of a Logging setup that uses all four of these objects.
+
+.. code-block:: python
+    LOGGING = {
+    "version": 1,
+    "disable_existing_loggers": False,
+    "formatters": {
+        "verbose": {
+            "format": "{levelname} {asctime} {module} {process:d} {thread:d} {message}",
+            "style": "{",
+        },
+        "simple": {
+            "format": "{levelname} {message}",
+            "style": "{",
+        },
+    },
+    "filters": {
+        "special": {
+            "()": "project.logging.SpecialFilter",
+            "foo": "bar",
+        },
+        "require_debug_true": {
+            "()": "django.utils.log.RequireDebugTrue",
+        },
+    },
+    "handlers": {
+        "console": {
+            "level": "INFO",
+            "filters": ["require_debug_true"],
+            "class": "logging.StreamHandler",
+            "formatter": "simple",
+        },
+        "mail_admins": {
+            "level": "ERROR",
+            "class": "django.utils.log.AdminEmailHandler",
+            "filters": ["special"],
+        },
+    },
+    "loggers": {
+        "django": {
+            "handlers": ["console"],
+            "propagate": True,
+        },
+        "django.request": {
+            "handlers": ["mail_admins"],
+            "level": "ERROR",
+            "propagate": False,
+        },
+        "myproject.custom": {
+            "handlers": ["console", "mail_admins"],
+            "level": "INFO",
+            "filters": ["special"],
+        },
+    },
+}
+
+=====================
+Logging GRPC Requests
+=====================
+
+Django-Socio-GRPC by default log requests only when something goes wrong just like Django does.
+    - If your service raises a GRPCException, it will result by default to a Warning.
+      You can define your own GRPCException and set the log_level as you wish to change this.
+    - If your service raises another Exception, it will result in an Error.
+    - Lastly, if you want to log requests even if they are ok, you can launch django in DEBUG mode
```

Comment:
replace by:
- Logging incoming request is only activated when in DEBUG mode or if LOG_OK_RESPONSE settings is set to True